core/io/resource.cpp

```diff
@@ -94,12 +95,43 @@ String Resource::get_path() const {
 	return path_cache;
 }
 
-void Resource::set_subindex(int p_sub_index) {
-	subindex = p_sub_index;
+String Resource::generate_scene_unique_id() {
+	// Generate a unique enough hash, but still user-readable.
+	// If it's not unique it does not matter because the saver will try again.
+	OS::Date date = OS::get_singleton()->get_date();
+	OS::Time time = OS::get_singleton()->get_time();
+	uint32_t hash = hash_djb2_one_32(OS::get_singleton()->get_ticks_usec());
+	hash = hash_djb2_one_32(date.year, hash);
+	hash = hash_djb2_one_32(date.month, hash);
+	hash = hash_djb2_one_32(date.day, hash);
+	hash = hash_djb2_one_32(time.hour, hash);
+	hash = hash_djb2_one_32(time.minute, hash);
+	hash = hash_djb2_one_32(time.second, hash);
+	hash = hash_djb2_one_32(Math::rand(), hash);
+
+	static const uint32_t characters = 5;
+	String id;
+	const uint32_t char_count = ('z' - 'a');
+	uint32_t base = char_count + ('9' - '0');
```

Comment:
```suggestion
	static constexpr uint32_t characters = 5;
	String id;
	static constexpr uint32_t char_count = ('z' - 'a');
	static constexpr uint32_t base = char_count + ('9' - '0');
```
Not sure if this is actually better, but it seems weird to me that `characters`, `char_count` and `base` are all constant but they all have different constant/static modifiers.